src/internal/sdata/sdata.go

```diff
@@ -124,3 +126,68 @@ func Copy(w TupleWriter, r TupleReader, row Tuple) (n int, _ error) {
 	}
 	return n, nil
 }
+
+func NewTupleFromTable(ctx context.Context, db *pachsql.DB, tableName string) (Tuple, error) {
+	info, err := pachsql.GetTableInfo(ctx, db, tableName)
+	if err != nil {
+		return nil, err
+	}
+	tuple := make(Tuple, len(info.Columns))
+	for i, ci := range info.Columns {
+		var err error
+		tuple[i], err = makeTupleElement(ci.DataType, ci.IsNullable)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return tuple, nil
+}
+
+func makeTupleElement(dbType string, nullable bool) (interface{}, error) {
+	switch dbType {
+	case "BOOL":
+		if nullable {
+			return &sql.NullBool{}, nil
```

Comment:
any reason to not use `new` for these, too? I don't care too much about which but it feels odd to mix and match in a single if statement

Reply:
Good point, I can change this to using `new`. I assume the difference is due to the fact that one is initializing a pointer, while the `new` is used for non-pointers?

Reply:
`new` still makes pointers, it's really just that you can't do the equivalent of `&structType{}` with primitives like `int`, `x := new(int)` is basically `var temp int; x := &temp`